File: 프로그래머스/lv3/장정윤/이중우선순위큐.py
def solution(operations):
    answer = []
    for opr in operations:
        ops = opr.split(' ')
        if ops[0] == 'I':
            answer.append(int(ops[1]))
            answer.sort()
        elif opr[0] == 'D' and len(answer) != 0:
            if int(ops[1]) == 1:
               answer.pop() 
            elif int(ops[1]) == -1:
               answer.pop(0)
    if len(answer) == 0:
        return [0,0]
    else:
        return [answer[-1], answer[0]]

File: 프로그래머스/lv3/김미래/이중우선순위 큐.cpp
#include <string>
#include <vector>
#include <list>
#include <algorithm>
#include <iostream>
using namespace std;

vector<int> solution(vector<string> operations) {
    vector<int> answer;
    list<int> q;
    
    for(int i=0; i<operations.size(); i++){
        if(operations[i][0] == 'I'){
            int num = stoi(operations[i].substr(2));
            q.push_front(num);
            q.sort();
        }else if(operations[i][0] == 'D'){
            if(q.empty()){
                continue;
            }else{
                if(operations[i][2] == '-'){
                    q.pop_front();
                }else{
                    q.pop_back();
                }
            }
        }
    }
    
    if(q.empty()){
        answer.push_back(0);
        answer.push_back(0);
    }else{
        answer.push_back(q.back());
        answer.push_back(q.front());
    }
    return answer;
}
